File: benchmark-php/app/Logger/ConsoleLogger.php
<?php

namespace Benchmark\Commands;


use Becnhmark\Logger\ILogger;
use Benchmark\Metrics\Info;
use Nette\Utils\DateTime;
use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use Symfony\Component\Console\Style\SymfonyStyle;

class ConsoleLogger implements ILogger
{
    /** @var SymfonyStyle */
    private $io;


    public function __construct(InputInterface $input, OutputInterface $output)
    {
        $this->io = new SymfonyStyle($input, $output);
    }

    public function startMessage(Info $info)
    {
		$timestamp = $this->getCurrentTimestamp();
        $this->io->text('[INFO] ' . $timestamp . ' Benchmark of "' . $info->getFullName() . '" has started.');
    }

    public function endMessage(Info $info)
    {
    	$timestamp = $this->getCurrentTimestamp();
        $this->io->text('[INFO] ' . $timestamp . ' Benchmark of "' . $info->getFullName() . '" has finished.');
    }

	/**
	 * @return string
	 */
    private function getCurrentTimestamp() {

    	return (new DateTime('now'))->format('Y-m-d H:i:s');
	}
}

File: benchmark-java/src/main/java/benchmark/java/logger/ConsoleLogger.java
package benchmark.java.logger;

import benchmark.java.metrics.Info;
import java.io.PrintStream;
import java.sql.Timestamp;


public class ConsoleLogger implements ILogger {
	
	private final PrintStream printStream;

	public ConsoleLogger(PrintStream printStream) {
		this.printStream = printStream;
	}
	
	@Override
	public void startMessage(Info info) {
		
		String timestamp = getCurrentTimestamp();
		printStream.println("[INFO] " + timestamp + " Benchmark of \"" + info.getFullName() + "\" has started.");
	}

	@Override
	public void endMessage(Info info) {
		
		String timestamp = getCurrentTimestamp();
		printStream.println("[INFO] " + timestamp + " Benchmark of \"" + info.getFullName() + "\" has finished.");
	}
	
	private String getCurrentTimestamp() {
		
		Timestamp timestamp = new Timestamp(System.currentTimeMillis());
		return timestamp.toString();
	}
}
